ambari-server/src/main/java/org/apache/ambari/server/controller/internal/UpgradeResourceProvider.java

```diff
@@ -1654,8 +1654,11 @@ private ActionExecutionContext buildActionExecutionContext(Cluster cluster,
     actionContext.setStackId(stackId);
     actionContext.setTimeout(timeout);
     actionContext.setRetryAllowed(allowRetry);
-    actionContext.setAutoSkipFailures(context.isComponentFailureAutoSkipped());
-
+    if (StageWrapper.Type.SERVICE_CHECK.name().equals(role)) {
```

Comment:
I wouldn't do this check here. I'd just set it directly in `makeServiceCheckStage(...)`.

Reply:
Actually, I'm OK with this change, but I'm worried that `makeServiceCheckStage(...)` takes a skip boolean too. We should try do set it all in one place.